symbolic-debuginfo/src/dwarf.rs

```diff
@@ -1320,6 +1344,31 @@ impl<'s> Iterator for DwarfUnitFileIterator<'s> {
     }
 }
 
+fn resolve_byte_name<'s>(bcsymbolmap: Option<&BcSymbolMap<'s>>, s: &'s [u8]) -> &'s [u8] {
+    if let Some(bcsymbolmap) = bcsymbolmap {
+        std::str::from_utf8(s)
+            .map(|s| bcsymbolmap.resolve(s))
+            .ok()
+            .map(|s| s.as_bytes())
+            .unwrap_or(s)
+    } else {
+        s
+    }
+}
+
+fn resolve_cow_name<'s>(bcsymbolmap: Option<&BcSymbolMap<'s>>, s: Cow<'s, str>) -> Cow<'s, str> {
+    if let Some(bcsymbolmap) = bcsymbolmap {
+        match s {
+            Cow::Borrowed(s) => bcsymbolmap.resolve(s).into(),
+            // we make an unconditional copy here, since the borrow checker has no idea
+            // if the returned `str` is owned by the `bcsymbolmap`, or by `s`.
+            Cow::Owned(s) => bcsymbolmap.resolve(&s).to_owned().into(),
+        }
+    } else {
+        s
+    }
+}
```

Comment:
I wonder if these should be on the `BcSymbolMap` struct.  but that would mean committing to a public API so perhaps it's fine to leave them here as helpers.  (If you want to go all out you could make a local extension trait, but meh)

Reply:
The convenience in these helpers is also that they take care of the `Option`. I simplified them a bit, but they are just too convenient to pass by :-D